cpp/include/cuml/cluster/dbscan.hpp

```diff
@@ -53,4 +54,43 @@ void dbscanFit(const cumlHandle &handle, double *input, int64_t n_rows,
 
 /** @} */
 
+/**
```

Comment:
Since both C and python API's are being changed to support this feature, maybe it also makes sense to change the C++ API and remove the other `dbscanFit` methods above?

Reply:
Agreed, maybe with `<type> *core_sample_indices = nullptr`

Reply:
Apart from that, I had a quick look and everything seems LGTM to me.

Reply:
I tried this but some of the existing uses of the `dbscanFit` method will not compile if they specified the `max_bytes_per_batch` argument. One example is here (taken from the build log):
```
cpp/examples/dbscan/dbscan_example.cpp:227:43: error: invalid conversion from ‘size_t {aka long unsigned int}’ to ‘int*’ [-fpermissive]
                 max_bytes_per_batch, false);
```

I added the additional overrides without the `core_sample_indices` argument to allow backwards compatibility with existing examples/tests.

Reply:
yes what @teju85 suggested would be a breaking change in the C++ API as well. The existing examples/tests would have to be adapted accordingly.
A compromise would be to add the new argument at the end, but I don't really like that solution either, because the order of arguments seems very arbitrary in that case.
IMO, it's best to make breaking changes as early as possible in development of a library, and since this is something which allows us to be closer to sklearn, I think it's OK to make that breaking change now.
Let's wait on what @teju85 says about this

Reply:
There aren't that many direct users of c++ api currently. So, probably better to make this change now itself.

Reply:
Ok, if we are fine with the breaking change, I will move fowrard with that. There aren't many places to clean up.